### PR TITLE
feat: Add integration test of tickSpacing>1

### PR DESCRIPTION
**Pull request by lnist**
Also adds coercion of tick arguments, removed parallel test run, and fixes updateOffer on LiquidityProvider for an EOA.

### CODE REVIEW

File: src/util/tickPriceHelper.ts [hunk 1]
```diff
@@ -246,6 +249,21 @@ class TickPriceHelper {
     return binnedTick;
   }
 
+  /** Calculates the tick offset closely corresponding to the given raw ratio.
+   * @param priceRatio the price ratio.
```
Comment (espendk) [inline src/util/tickPriceHelper.ts:253]: What is a "price ratio"?

File: src/util/trade.ts [hunk 2]
```diff
@@ -62,26 +62,16 @@ class Trade {
...
+      maxTick = params.maxTick;
       if (slippage > 0) {
-        // round down to not exceed the price when buying, and up to get at least price for when selling
-        const limitPrice = tickPriceHelper.priceFromTick(
-          params.maxTick,
-          bs === "buy" ? "roundDown" : "roundUp",
+        // offset the tick by getting a non-rounded offset
+        maxTick += tickPriceHelper.tickOffsetFromRawRatio(
+          Big(1).sub(Big(100 + bs === "buy" ? slippage : -slippage).div(100)),
+          "none",
```
Comment (espendk) [inline src/util/trade.ts:70]: If this is the only use for `tickOffsetFromRawRatio`, then maybe we could replace it with a method that takes a tick and the percentage that should be added to it? Or something like that.

Right now, the method itself and the call site are somewhat difficult to understand. Would be good if the complexity was only in one place :-)

**Review by espendk — COMMENTED** [review]
nice!

I'm unsure of some of the terminology/naming, I think it could be clearer. Let's talk about it.

File: src/util/tickPriceHelper.ts [hunk 3]
```diff
@@ -208,17 +208,20 @@ class TickPriceHelper {
...
     outboundVolume: Bigish,
-    roundingMode: RoundingMode,
+    roundingMode: RoundingMode | "none",
   ): number {
     const rawInbound = this.#getInbound().toUnits(inboundVolume);
     const rawOutbound = this.#getOutbound().toUnits(outboundVolume);
     const tick = TickLib.tickFromVolumes(rawInbound, rawOutbound);
+    if (roundingMode === "none") {
+      return tick.toNumber();
+    }
```
[the remark this answers is not in the document]
Comment (lnist) [reply src/util/tickPriceHelper.ts:224]: Removed the option.

File: src/util/tickPriceHelper.ts (hunk 1 above)
Comment (lnist) [reply src/util/tickPriceHelper.ts:253]: Concept from kandel - removed from here as function is deleted.

File: src/util/trade.ts (hunk 2 above)
Comment (lnist) [reply src/util/trade.ts:70]: Replaced by usage of tickFromRawRatio and simplified expression.

File: src/util/tickPriceHelper.ts [hunk 4]
```diff
@@ -246,6 +249,21 @@ class TickPriceHelper {
     return binnedTick;
   }
 
+  /** Calculates the tick offset closely corresponding to the given raw ratio.
+   * @param priceRatio the price ratio.
+   * @param roundingMode See {@link RoundingMode}. roundDown is to a lower tick, roundUp is to a higher tick. none is no rounding.
+   * @returns The tick offset corresponding to the price ratio.
+   */
+  tickOffsetFromRawRatio(priceRatio: Big, roundingMode: RoundingMode | "none") {
```
[the remark this answers is not in the document]
Comment (lnist) [reply src/util/tickPriceHelper.ts]: It is meant to mean a difference between two ticks which yields the equivalent ratio difference between their ratios. But function is deleted.

**Review by espendk — APPROVED** [review]
Nice improvements 👌

**Pull request merged**